src/vm/coreassemblyspec.cpp

```diff
@@ -170,6 +170,9 @@ VOID  AssemblySpec::Bind(AppDomain      *pAppDomain,
         hr = CreateAssemblyNameObject(&pName, assemblyDisplayName, CANOF_PARSE_DISPLAY_NAME, NULL);
         if (SUCCEEDED(hr))
         {
+            Holder<Thread*, DoNothing, Thread::ResetParentOfLoadAssemblyByName> holder(GetThread());
+            if(GetThread() != NULL)
+            	GetThread()->SetParentOfLoadAssemblyByName(GetParentAssembly());
```

Comment:
I concur with @jkotas on this. You should consider passing the parent assembly reference to the binder that will eventually work its way back for you to have access to it.